File: Longest Possible Route in a Matrix with Hurdles.java
//Problem: Longest Possible Route in a Matrix with Hurdles
//geeksforgeeks problem of the day -> 11th July 2026
//JAVA code
class Solution {
    int n, m;
    boolean[][] visited;
    int[] dr = {-1, 1, 0, 0};
    int[] dc = {0, 0, -1, 1};
    
    public int longestPath(int[][] mat, int xs, int ys, int xd, int yd) {
        n = mat.length;
        m = mat[0].length;
        
        if(mat[xs][ys] == 0 || mat[xd][yd] == 0){
            return -1;
        }
        visited = new boolean[n][m];
        return dfs(mat, xs, ys, xd, yd);
    }
    
    private int dfs(int[][] mat, int r, int c, int xd, int yd){
        if(r == xd && c == yd){
            return 0;
        }
        visited[r][c] = true;
        int maxLen = -1;
        
        for(int k=0;k<4;k++){
            int nr = r+dr[k];
            int nc = c+dc[k];
            
            if(isSafe(mat, nr, nc)){
                int path = dfs(mat, nr, nc, xd, yd);
                if(path != -1){
                    maxLen = Math.max(maxLen, path+1);
                }
            }
        }
        visited[r][c] = false;
        return maxLen;
    }
    
    private boolean isSafe(int[][] mat, int r, int c){
        return r>=0 && r<n && c>=0 && c<m && mat[r][c] == 1 && !visited[r][c];
    }
}

File: Longest Possible Route in a Matrix with Hurdles.cpp
//Problem: Longest Possible Route in a Matrix with Hurdles
//geeksforgeeks problem of the day -> 11th July 2026
//C++ code
class Solution {
  public:
    int longestPath(vector<vector<int>> matrix, int xs, int ys, int xd, int yd)
    {
        // code here
        
        if(matrix[xs][ys]==0 || matrix[xd][yd]==0)return -1;
        int ans = -1;
        findLongestPath(matrix,xs,ys,xd,yd,0,ans);
        return ans;
        
    }
    
    void findLongestPath(vector<vector<int>> &matrix,int x,int y,int xd,int yd,int pathlength,int &ans)
     {
         if(x == xd && y == yd)
         {  if(pathlength>ans)ans = pathlength; return; }
         
         matrix[x][y]  = 2; 
         
         vector<pair<int,int>>neighbours = {{0,1}, {0,-1}, {1,0} ,{-1,0}};
         
         for(pair<int,int>neighbour : neighbours) {
             int newx = neighbour.first+x , newy = neighbour.second+y;
         if(newx>=0 && newy>=0 && newx<matrix.size() && newy<matrix[0].size() && matrix[newx][newy]==1)
             findLongestPath(matrix,newx,newy,xd,yd,pathlength+1,ans); }
         
         matrix[x][y] =1;
     }
};